apps/meteor/app/lib/server/functions/sendMessage.ts

```diff
@@ -215,11 +216,28 @@ export function prepareMessageObject(
 /**
  * Validates and sends the message object.
  */
-export const sendMessage = async function (user: any, message: any, room: any, upsert = false, previewUrls?: string[]) {
+export const sendMessage = async (
+	user: any,
+	message: any,
+	room: any,
+	upsert = false,
+	previewUrls?: string[],
+	uploadIdsToConfirm?: string[],
+) => {
 	if (!user || !message || !room._id) {
 		return false;
 	}
 
+	if (uploadIdsToConfirm !== undefined) {
+		const filesToConfirm: Partial<IUpload>[] = await Uploads.findByIds(uploadIdsToConfirm).toArray();
```

Comment:
Do not load data if it's not gonna be used. Load it inside the condition that uses it instead.

Reply:
Done